crates/fluvio-service/src/server.rs

```diff
@@ -62,159 +39,108 @@ pub trait FlvService {
-
-                        let peer = address.to_string();
-
-                        info!(server = %host,%peer, "new peer connection");
-
-                        let socket_res = builder.to_socket(stream);
-                        match socket_res.await {
-                            Ok(socket) => {
-                                let connection_info = ConnectInfo {
-                                    peer: peer.clone(),
-                                    host: host.clone(),
-                                };
-
-                                if let Err(err) = service
-                                    .respond(context.clone(), socket, connection_info)
-                                    .await
-                                {
-                                    error!(
-                                        "error handling stream: {}, shutdown on: {} from: {}",
-                                        err, host, address,
-                                    );
-                                } else {
-                                    info!(%host,%peer, "connection terminated");
-                                }
-                            }
-                            Err(err) => {
-                                error!(
-                                    "error on tls handshake: {}, on: {} from: addr: {}",
-                                    err, host, peer
-                                );
-                            }
-                        }
-                    };
-
-                    spawn(ft);
+                    spawn(Self::handle_request(stream, context, service, host));
                 }
-                Err(err) => {
-                    error!("error with stream: {}", err);
+                Err(e) => {
+                    error!("Error from TCP Stream: {:?}", e);
                 }
             }
-        } else {
-            trace!("no stream value, ignoring");
+        }
+
+        info!("Closed TcpListener");
+    }
+
+    #[instrument(skip(stream, context, service))]
+    async fn handle_request(stream: TcpStream, context: C, service: Arc<S>, host: String) {
+        let peer_addr = stream
+            .peer_addr()
+            .map(|addr| addr.to_string())
+            .unwrap_or_else(|_| "".to_owned());
+        debug!(%peer_addr, "Handling request");
+
+        let socket = {
+            let fd = stream.as_raw_fd();
+            FluvioSocket::from_stream(Box::new(stream.clone()), Box::new(stream), fd)
+        };
+
+        let connection_info = ConnectInfo {
+            peer: peer_addr.clone(),
+            host: host.clone(),
+        };
+
+        let result = service.respond(context, socket, connection_info).await;
+        match result {
+            Ok(_) => {
+                info!(%host, %peer_addr, "Response sent successfully, closing connection");
+            }
+            Err(err) => {
+                error!(%host, %peer_addr, "Failed to send response: {}", err);
```

Comment:
let's keep old log message.   `response` doesn't send response. it just handles request.

Reply:
Got it. What would you think of changing `respond` to `handle` then?